lib/ignore/tempname.go

```diff
@@ -33,33 +28,39 @@ const maxFilenameLength = 160 - len(unixTempPrefix) - len(".tmp")
 
 func init() {
 	if runtime.GOOS == "windows" {
-		defTempNamer = tempNamer{windowsTempPrefix, []string{unixTempPrefix, windowsTempPrefix}}
+		tempPrefix = windowsTempPrefix
 	} else {
-		defTempNamer = tempNamer{unixTempPrefix, []string{unixTempPrefix, windowsTempPrefix}}
+		tempPrefix = unixTempPrefix
 	}
 }
 
 // IsTemporary is true if the file name has the temporary prefix. Regardless
 // of the normally used prefix, the standard Windows and Unix temp prefixes
 // are always recognized as temp files.
-func (t tempNamer) IsTemporary(name string) bool {
+func IsTemporary(name string) bool {
 	name = filepath.Base(name)
-	for _, prefix := range t.recognize {
-		if strings.HasPrefix(name, prefix) {
-			return true
-		}
+	if strings.HasPrefix(name, windowsTempPrefix) ||
+		strings.HasPrefix(name, unixTempPrefix) {
+		return true
 	}
 	return false
 }
 
-func (t tempNamer) TempName(name string) string {
+func TempName(name string) string {
 	tdir := filepath.Dir(name)
 	tbase := filepath.Base(name)
 	if len(tbase) > maxFilenameLength {
 		hash := md5.New()
 		hash.Write([]byte(name))
 		tbase = fmt.Sprintf("%x", hash.Sum(nil))
 	}
-	tname := fmt.Sprintf("%s%s.tmp", t.prefix, tbase)
+	tname := fmt.Sprintf("%s%s.tmp", tempPrefix, tbase)
 	return filepath.Join(tdir, tname)
 }
+
+// SetWindowsPrefix is entirely reserved for testing, it should never be called
+// in production code.
+// It is needed to allow static test files independent from underlying os.
+func SetWindowsPrefix() {
```

Comment:
Looks good to me except for this method which is ugly and clobbers any other usage of temp files after usage, in tests by other packages for example. Better alternatives imho are exporting the variables themselves, or a more general setprefixes method that can be called before and after tests to restor.